### PR TITLE
Adds circular loader

### LINKED ISSUE #1584: Update animations for LoadingIndicator widget
**Enhancement** 

It would be helpful for a developer to have a loading widget that has a property value that would change the type of spinner animation as needed. The update could involve adding the property value of `type` to the `LoadingIndicator` widget that could have the basic values `linear` and `circular` as described here: https://material.io/components/progress-indicators#types as well as some others. Some examples I found are:

- https://spin.js.org/
- https://loading.io/css/
- https://ant.design/components/spin/

### PR DESCRIPTION
**Pull request by samends**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit tests are included in the PR
* [x] For new widgets, an entry has been added to the `.dojorc`
* [x] For new widgets, `theme.variant()` is added to the root domnode
* [x] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [x] WidgetProperties are exported

**Description:**
Adds circular loader to `LoadingIndicator` widget

Resolves #1584

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -69,6 +69,7 @@
     "@material/card": "5.1.0",
     "@material/checkbox": "5.1.0",
     "@material/chips": "5.1.0",
+    "@material/circular-progress": "6.0.0",
```
Comment (samends) [inline package.json:72]: version `5.1.0` doesn't exist for this package https://www.npmjs.com/package/@material/circular-progress

File: src/theme/default/loading-indicator.m.css [hunk 2]
```diff
@@ -21,3 +29,50 @@
...
+/*  An indeterminate SVG (there are three of these in total). */
+.circleGraphic {
+}
+
+/* A tiny little sliver of an SVG circle used to patch a gap between the circle-left and the circle-right. */
+.gapPatch {
+}
+
+/* Small size for circular loader */
+.small {
```
Comment (samends) [inline src/theme/default/loading-indicator.m.css:63]: sizing matches material ui https://github.com/material-components/material-components-web/tree/master/packages/mdc-circular-progress#sizing

File: src/examples/src/widgets/loading-indicator/Circular.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import Example from '../../Example';
+
+const factory = create({ icache });
+
+export default factory(function Circular({ middleware: { icache } }) {
+	const active = icache.getOrSet('active', true);
+	return (
+		<Example>
+			<virtual>
+				<LoadingIndicator type={'circular-small'} active={active} />
```
Comment (agubler) [inline src/examples/src/widgets/loading-indicator/Circular.tsx:14]:
```suggestion
				<LoadingIndicator type="circular-small" active={active} />
```

File: src/loading-indicator/index.tsx [hunk 4]
```diff
@@ -6,29 +6,113 @@ import { create, tsx } from '@dojo/framework/core/vdom';
 export interface LoadingIndicatorProperties extends ThemedProperties {
 	/** If the element is actively loading. Defaults to true */
 	active?: boolean;
+	/* Sets loader type to linear or circular */
```
Comment (agubler) [inline src/loading-indicator/index.tsx:9]:
```suggestion
	/* Sets loader type to linear or circular, defaults to `linear` */
```

File: src/loading-indicator/index.tsx [hunk 5]
```diff
@@ -6,29 +6,113 @@ import { create, tsx } from '@dojo/framework/core/vdom';
...
 
 export const LoadingIndicator = factory(function LoadingIndicator({
 	properties,
 	middleware: { theme }
 }) {
 	const classes = theme.classes(css);
-	const { active = true } = properties();
+	const { active = true, type = 'linear' } = properties();
+
+	const [loadingType, size = 'small'] =
```
Comment (agubler) [inline src/loading-indicator/index.tsx:24]: Can we not do this, it's better to be more explicit/verbose 😄 

```tsx
const isLinear = type === 'linear';

if (isLinear) {
    return (
        // the existing loading indicator
    )
}

let sizeClass: string | undefined;
switch (type) {
	case 'circular-small':
		sizeClass = themeCss.small;
		break;
	case 'circular-large':
		spacingClass = themeCss.large;
		break;
	case 'circular-medium':
	default:
		sizeClass = themeCss.medium;
		break;
}

return (
    // The circular indicator using the sizeClass
)
```

File: src/loading-indicator/index.tsx [hunk 6]
```diff
@@ -6,29 +6,113 @@ import { create, tsx } from '@dojo/framework/core/vdom';
...
+						classes.linear
+					]}
+					role="progressbar"
+				>
+					<div classes={classes.buffer} />
+					{active ? (
+						<div classes={[classes.bar, classes.primary]}>
+							<span classes={classes.inner} />
+						</div>
+					) : null}
```
Comment (agubler) [inline src/loading-indicator/index.tsx:45]:
```suggestion
					{active && (
						<div classes={[classes.bar, classes.primary]}>
							<span classes={classes.inner} />
						</div>
					)}
```

File: src/loading-indicator/index.tsx [hunk 7]
```diff
@@ -6,29 +6,113 @@ import { create, tsx } from '@dojo/framework/core/vdom';
...
+											r="8.75"
+											stroke-dasharray="54.978"
+											stroke-dashoffset="27.489"
+											stroke-width="2.5"
+										/>
+									</svg>
+								</div>
+							</div>
+						</div>
+					) : null}
```
Comment (agubler) [inline src/loading-indicator/index.tsx:112]:
```suggestion
					{active && (
						<div classes={classes.circularContainer}>
							<div classes={classes.spinnerLayer}>
								<div classes={[classes.circleClipper, classes.circleLeft]}>
									<svg
										classes={classes.circleGraphic}
										viewBox="0 0 24 24"
										xmlns="http://www.w3.org/2000/svg"
									>
										<circle
											cx="12"
											cy="12"
											r="8.75"
											stroke-dasharray="54.978"
											stroke-dashoffset="27.489"
											stroke-width="2.5"
										/>
									</svg>
								</div>
								<div classes={classes.gapPatch}>
									<svg
										classes={classes.circleGraphic}
										viewBox="0 0 24 24"
										xmlns="http://www.w3.org/2000/svg"
									>
										<circle
											cx="12"
											cy="12"
											r="8.75"
											stroke-dasharray="54.978"
											stroke-dashoffset="27.489"
											stroke-width="2"
										/>
									</svg>
								</div>
								<div classes={[classes.circleClipper, classes.circleRight]}>
									<svg
										classes={classes.circleGraphic}
										viewBox="0 0 24 24"
										xmlns="http://www.w3.org/2000/svg"
									>
										<circle
											cx="12"
											cy="12"
											r="8.75"
											stroke-dasharray="54.978"
											stroke-dashoffset="27.489"
											stroke-width="2.5"
										/>
									</svg>
								</div>
							</div>
						</div>
					)}
```

File: src/theme/dojo/loading-indicator.m.css [hunk 8]
```diff
@@ -1,4 +1,7 @@
 .root {
```
Comment (agubler) [inline src/theme/dojo/loading-indicator.m.css:1]: Should the dojo theme have the size classes with the appropriate CSS Vars?

File: src/theme/material/loading-indicator.m.css [hunk 9]
```diff
@@ -1,4 +1,4 @@
-.root {
+.linear {
```
Comment (agubler) [inline src/theme/material/loading-indicator.m.css:1]: Should the material theme have the size classes with the appropriate CSS Vars?

**Review by agubler — CHANGES_REQUESTED** [review]

File: src/loading-indicator/index.tsx (hunk 5 above)
Comment (samends) [reply src/loading-indicator/index.tsx:24]: Yea I was planning ahead too much, figured with that code we could add more types like `spin-small` without having to update the code. However over engineering for the future is thing

File: src/theme/dojo/loading-indicator.m.css (hunk 8 above)
Comment (samends) [reply src/theme/dojo/loading-indicator.m.css:1]: I've added the `.small` `.medium` and `.large` classes to the dojo and material theme, however should the sizes differ from the default sizing?

Comment (agubler) [reply src/theme/dojo/loading-indicator.m.css:1]: As long as they use CSS variables and they are configured in the variants for dojo/material default and dark then they can be the same as the default sizes - that will allow users to create a custom variant and change them

File: src/theme/dojo/loading-indicator.m.css [hunk 10]
```diff
@@ -7,6 +10,15 @@
 	width: 100%;
 }
 
+.circular {
+	border: calc(var(--grid-base) / 2) solid var(--color-background-faded);
+	border-radius: 50%;
+	border-top: calc(var(--grid-base) / 2) solid var(--color-background-inverted);
+	width: calc(var(--grid-base) * 6.25);
```
Comment (tomdye) [inline src/theme/dojo/loading-indicator.m.css:17]: I'm not sure that these sizes here etc are relevant to the grid size, ie. if the grid size changed would you expect the loader to change size?
Also, we should really refrain from fractional multipliers like these. Just hardcode the width / height appropriately  in this case.

Comment (samends) [reply src/theme/dojo/loading-indicator.m.css]: Removed the width/height here as they're not needed

**Comment by tomdye** [comment]
@samends Looks like you've addressed Ant's concerns but I believe it would benefit from having the different sizes examples added to the circular indicator example please.

**Review by tomdye — APPROVED** [review]
Thanks for the extra examples 👍

**Pull request merged**